Assets/MixedRealityToolkit/_Core/Devices/MixedReality/WindowsMixedRealityController.cs

```diff
@@ -15,270 +14,453 @@
+                switch (mappings[i].AxisType)
+                {
+                    case AxisType.Digital:
+                        interactions.Add(new MixedRealityInteractionMapping<bool>(i, mappings[i].AxisType, mappings[i].InputType, mappings[i].InputAction));
+                        break;
+                    case AxisType.SingleAxis:
+                        interactions.Add(new MixedRealityInteractionMapping<float>(i, mappings[i].AxisType, mappings[i].InputType, mappings[i].InputAction));
+                        break;
+                    case AxisType.DualAxis:
+                        interactions.Add(new MixedRealityInteractionMapping<Vector2>(i, mappings[i].AxisType, mappings[i].InputType, mappings[i].InputAction));
+                        break;
+                    case AxisType.ThreeDofPosition:
+                        interactions.Add(new MixedRealityInteractionMapping<Vector3>(i, mappings[i].AxisType, mappings[i].InputType, mappings[i].InputAction));
+                        break;
+                    case AxisType.ThreeDofRotation:
+                        interactions.Add(new MixedRealityInteractionMapping<Quaternion>(i, mappings[i].AxisType, mappings[i].InputType, mappings[i].InputAction));
+                        break;
+                    case AxisType.SixDof:
+                        interactions.Add(new MixedRealityInteractionMapping<SixDof>(i, mappings[i].AxisType, mappings[i].InputType, mappings[i].InputAction));
+                        break;
+                    case AxisType.None:
+                    case AxisType.Raw:
+                    default:
+                        interactions.Add(new MixedRealityInteractionMapping<object>(i, mappings[i].AxisType, mappings[i].InputType, mappings[i].InputAction));
+                        break;
+                }
             }
+            Interactions = interactions.ToArray();
         }
 
         /// <summary>
         /// Create Interaction mappings from a device specific default set of action mappings
         /// </summary>
         /// <param name="interactionSourceState">The InteractionSourceState retrieved from the platform</param>
-        private void SetupWMRControllerDefaults(InteractionSourceState interactionSourceState)
+        private void SetupWMRControllerDefaults()
         {
+            var interactions = new List<IMixedRealityInteractionMapping>();
+
             InputAction[] inputActions = Managers.MixedRealityManager.Instance.ActiveProfile.InputActionsProfile.InputActions;
             if (inputActions == null)
             {
                 return;
             }
+            //Add the Controller Pointer
+            interactions.Add(new MixedRealityInteractionMapping<SixDof>(1, AxisType.SixDof, DeviceInputType.SpatialPointer, new InputAction(1, "Select"))); // Note will convert these lookups to indexes
```

Comment:
We should probably be getting this data from Managers.MixedRealityManager.Instance.ActiveProfile.ControllersProfile.MixedRealityControllerMappingProfiles

Reply:
it does, that is what the "SetupFromMapping" function is for.  The defaults are only for when there are NO mappings at all configured

Reply:
We wouldn't have any mappings configured if we're not using the input system correct?

Reply:
Err, no. The input system is meant to be "optional" :P 
#FoolofaTook :P